File: VB/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/ImmediatePostDataDemo.vb
Imports DevExpress.Persistent.Base
Imports DevExpress.Persistent.BaseImpl
Imports DevExpress.Xpo

<DefaultClassOptions>
Public Class ImmediatePostDataDemo
    Inherits BaseObject

    Public Sub New(ByVal session As Session)
        MyBase.New(session)
    End Sub

    Private _lastName As String

    <ImmediatePostData>
    Public Property LastName As String
        Get
            Return _lastName
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(LastName), _lastName, value)
        End Set
    End Property

    Private _firstName As String

    <ImmediatePostData>
    Public Property FirstName As String
        Get
            Return _firstName
        End Get
        Set(ByVal value As String)
            SetPropertyValue(Of String)(NameOf(FirstName), _firstName, value)
        End Set
    End Property

    Public ReadOnly Property FullName As String
        Get
            Return $"{LastName} {FirstName}"
        End Get
    End Property

    Private _booleanProperty As Boolean

    <ImmediatePostData>
    Public Property BooleanProperty As Boolean
        Get
            Return _booleanProperty
        End Get
        Set(ByVal value As Boolean)
            SetPropertyValue(Of Boolean)(NameOf(BooleanProperty), _booleanProperty, value)
        End Set
    End Property

    Public ReadOnly Property BooleanPropertyAsString As String
        Get
            Return $"{BooleanProperty.ToString()}"
        End Get
    End Property

    Private _numericProperty As Integer

    <ImmediatePostData>
    Public Property NumericProperty As Integer
        Get
            Return _numericProperty
        End Get
        Set(ByVal value As Integer)
            SetPropertyValue(Of Integer)(NameOf(NumericProperty), _numericProperty, value)
        End Set
    End Property

    Public ReadOnly Property SquaredNumericProperty As Integer
        Get
            Return NumericProperty * NumericProperty
        End Get
    End Property
End Class

File: CS/EditReferenceProperiesInBatchEditMode.Module/BusinessObjects/ImmediatePostDataDemo.cs
using DevExpress.Persistent.Base;
using DevExpress.Persistent.BaseImpl;
using DevExpress.Xpo;

namespace EditReferenceProperiesInBatchEditMode.Module.BusinessObjects {
    [DefaultClassOptions]
    public class ImmediatePostDataDemo : BaseObject {
        public ImmediatePostDataDemo(Session session) : base(session) { }
        private string lastName;
        [ImmediatePostData]
        public string LastName {
            get { return lastName; }
            set { SetPropertyValue<string>(nameof(LastName), ref lastName, value); }
        }
        private string firstName;
        [ImmediatePostData]
        public string FirstName {
            get { return firstName; }
            set { SetPropertyValue<string>(nameof(FirstName), ref firstName, value); }
        }
        public string FullName {
            get { return $"{LastName} {FirstName}"; }
        }
        private bool booleanProperty;
        [ImmediatePostData]
        public bool BooleanProperty {
            get { return booleanProperty; }
            set { SetPropertyValue<bool>(nameof(BooleanProperty), ref booleanProperty, value); }
        }
        public string BooleanPropertyAsString {
            get { return $"{BooleanProperty.ToString()}"; }
        }
        private int numericProperty;
        [ImmediatePostData]
        public int NumericProperty {
            get { return numericProperty; }
            set { SetPropertyValue<int>(nameof(NumericProperty), ref numericProperty, value); }
        }
        public int SquaredNumericProperty {
            get { return NumericProperty * NumericProperty; }
        }
    }
}
